x/stake/keeper/validator.go

```diff
@@ -210,19 +212,26 @@ func (k Keeper) UpdateValidator(ctx sdk.Context, validator types.Validator) type
 	cliffPower := k.GetCliffValidatorPower(ctx)
 
 	switch {
-	// if already bonded and power increasing only need to update tendermint
+	// if the validator is already bonded and the power is increasing, we need
+	// perform the following:
+	// a) update Tendermint
+	// b) check if the cliff validator needs to be updated
 	case powerIncreasing && !validator.Revoked &&
 		(oldFound && oldValidator.Status == sdk.Bonded):
 
 		bz := k.cdc.MustMarshalBinary(validator.ABCIValidator())
 		store.Set(GetTendermintUpdatesKey(validator.Owner), bz)
 
+		if cliffPower != nil {
+			k.attemptUpdateCliffValidator(ctx, validator)
```

Comment:
Why don't we just call this `UpdateCliffValidator` and move the code currently in this `attempt` function to the switch statement here: 

```
if cliffPower != nil {
	cliffAddr := sdk.AccAddress(k.GetCliffValidator(ctx))
	if bytes.Equal(cliffAddr, affectedVal.Owner) {
		k.UpdateCliffValidator(ctx, validator)
	}
}
```

I bet there is an even nicer way to combine those if statements too

Reply:
I like this idea as well. I'll update.